plugin-copilot/src/main/java/appland/copilotChat/copilot/GitHubCopilotService.java

```diff
@@ -0,0 +1,144 @@
+    }
+
+    /**
+     * Creates a new chat session with the GitHub Copilot API.
+     * If the OAuth token or the Copilot token could not be retrieved, this method will return {@code null}.
+     */
+    public @Nullable CopilotChatSession createChatSession() {
+        var githubToken = loadGitHubOAuthToken();
+        if (githubToken == null) {
+            return null;
+        }
+
+        var copilotToken = UpdatingCopilotToken.fetch(githubToken);
+        if (copilotToken == null) {
+            return null;
+        }
+
+        var apiEndpoint = copilotToken.getToken().endpoints().getOrDefault(CopilotToken.CopilotEndpoint.API, "https://api.githubcopilot.com");
+        var baseHeaders = Map.of(
+                "copilot-language-server-version", GitHubCopilot.LANGUAGE_SERVER_VERSION,
+                "editor-plugin-version", GitHubCopilot.GITHUB_COPILOT_PLUGIN_VERSION,
+                "editor-version", getCopilotEditorVersion(),
+                "user-agent", GitHubCopilot.USER_AGENT,
+                "x-github-api-version", GitHubCopilot.GITHUB_API_VERSION,
+                "vscode-machineid", machineId,
+                "vscode-sessionid", UUID.randomUUID().toString()
+        );
+        return new CopilotChatSession(apiEndpoint, copilotToken, baseHeaders);
+    }
+
+    /**
+     * @return The value of the "oauth_token" field in the GitHub Copilot configuration file, or {@code null} if it could not be found.
+     */
+    private @Nullable String loadGitHubOAuthToken() {
+        var configPath = findGitHubCopilotConfig();
+        if (configPath == null) {
+            LOG.debug("Unable to find an existing GitHub Copilot configuration file.");
+            return null;
+        }
+
+        try {
+            LOG.debug("Reading GitHub Copilot config: " + configPath);
+
+            var config = GsonUtils.GSON.fromJson(Files.readString(configPath, StandardCharsets.UTF_8), JsonObject.class);
+            for (var entry : config.asMap().entrySet()) {
+                if (entry.getKey().startsWith("github.com") && entry.getValue().isJsonObject()) {
+                    var oauthToken = entry.getValue().getAsJsonObject().getAsJsonPrimitive("oauth_token");
+                    if (oauthToken != null) {
+                        return oauthToken.getAsString();
+                    }
+                }
+            }
+
+            LOG.debug("No GitHub OAuth token found in GitHub Copilot config: " + configPath);
+            return null;
+        } catch (IOException e) {
+            LOG.debug("Failed to read GitHub Copilot config: " + configPath, e);
+            return null;
+        }
+    }
+
+    /**
+     * @return The {@link Path} to the GitHub Copilot plugin's configuration file, or {@code null} if it could not be found.
+     */
+    private static @Null Path findGitHubCopilotConfig() {
+        var basePath = Path.of(SystemProperties.getUserHome(), ".config", "github-copilot");
```

Comment:
Will this work on Windows?

Reply:
It should, but I'll test again on Windows now.

Reply:
@dividedmind Thanks for raising this, it did not work on Windows 11. I'll fix this now.